doc/crypto/api/ops/encap.rst

```diff
@@ -0,0 +1,326 @@
+Key Encapsulation
+=================
+
+Elliptic Curve Integrated Encryption Scheme
+-------------------------------------------
+
+The Elliptic Curve Integrated Encryption Scheme was fist proposed by Shoup, then improved by Ballare and Rogaway.
+
+The original specification permitted a number of variants. This specification only defines the version specified in [SEC1], that is with the use of labels and with the label size defined in bytes. 
+
+It is possible that some applications may need to use older versions to interoperate with legacy systems. 
+While the application can always implement this using the other algorithm functions provided, however, an implementation may choose to add these as a convenience in the implementation numbering space. 
+
+.. macro:: PSA_ALG_ECIES

```

Comment:
Do we want/need to provide separate algorithms for ECIES with ECDH, and for ECIES with EC-Cofactor-DH? (as described in SEC1). I assuming that this cannot be deduced from the key attributes.

Reply:
PSA_ALG_ECIES references Sec 1 so is cofactor. 

We possibly should call it PSA_ALG_ECIES_SEC1

We could add PSA_ALG_ECIES_18033 to cover the ISO version without cofactors. 
However, in earlier discussion in the Tuesday calls, Giles suggested we just define one for now.

It would be trivial to add PSA_ALG_ECIES_18033

Reply:
SEC1 v2.0 describes ECIES using either ECDH or EC-Cofactor-DH. See §5.1.1 point 4 (page 51 in https://www.secg.org/sec1-v2.pdf), which has this choice as a selected option/parameter by the participants.

However, `PSA_ALG_ECDH` in the Crypto API is described as implementing the cofactor computation. See the last few paragraphs in the [description of `PSA_ALG_ECDH`](https://arm-software.github.io/psa-api/crypto/1.2/api/ops/ka.html#c.PSA_ALG_ECDH). So it is reasonable for `PSA_ALG_ECIES` to be defined as using the same key agreement computation.